### PR TITLE
SWARM-1926 writer review

### PR DESCRIPTION
**Pull request by tradej**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: docs/reference/configuration.adoc [hunk 1]
```diff
@@ -22,7 +22,7 @@ In practical usage, the property would be, for example, `swarm.undertow.servers.
 ====
 
 [discrete]
-=== Setting System Properties using Maven Plugin
+=== Setting system properties using Maven plugin
```
Comment (tradej) [inline docs/reference/configuration.adoc:25]: I know we are not 100% consistent about this in the docs, but the consensus is to move to sentence case (Only the first letter of the first word is uppercase) as opposed to the historically prevalent title case

File: docs/reference/configuration.adoc [hunk 2]
```diff
@@ -165,47 +165,81 @@ $ java -jar myapp-swarm.jar -Scloud -Stesting -s/home/app/openshift.yml
 ----
 ====
 
-[#configuring-an-application-using-env]
-== Configuring an application using Environment Variables
+[#configuring-an-application-using-environment-variables]
```
Comment (tradej) [inline docs/reference/configuration.adoc:168]: Anchors should follow the headings as close as possible.

**Comment by rareddy** [comment]
Thanks @tradej I will remember for next time :) Also I know I put in data source example here, but do you think it is useful to even have the same in "data sources" doc page?

File: docs/reference/configuration.adoc [hunk 3]
```diff
@@ -165,47 +165,81 @@ $ java -jar myapp-swarm.jar -Scloud -Stesting -s/home/app/openshift.yml
...
-Linux based container Rules
-====
-* It is a naming convention that all environment properties are defined using UPPERCASE letters. For example: property `serveraddress` is defined as `SERVERADDRESS`. 
-* All the dot(.) characters MUST to be replaced with underscore (_). For example: property `swarm.bind.address=127.0.0.1` should be defined as `SWARM_BIND_ADDRESS=127.0.0.1`
-* All dash(-) characters MUST be replaced with string "__DASH__". For example: a property `swarm.data-sources.foo.url=<url>` needs to be `SWARM_DATA_DASH_SOURCES_FOO_URL=<url>`
-* If the property name has underscores then all underscore(-) characters MUST be replaced with string "__UNDERSCORE__". For example: a property `swarm.data_sources.foo.url=<url>` needs to be `SWARM_DATA_UNDERSCORE_SOURCES_FOO_URL=<url>`
-=====
+.Linux-based container rules
+* It is a naming convention that all environment properties are defined using uppercase letters.
+For example, define the `serveraddress` property as `SERVERADDRESS`.
```
Comment (tradej) [inline docs/reference/configuration.adoc:188]: The convention is to use one sentence per line.

File: docs/reference/configuration.adoc [hunk 4]
```diff
@@ -165,47 +165,81 @@ $ java -jar myapp-swarm.jar -Scloud -Stesting -s/home/app/openshift.yml
...
+For example, define the `swarm.data_sources.foo.url=<url>` property as `SWARM_DATA_UNDERSCORE_SOURCES_FOO_URL=<url>`.
 
-Example Data Source Configuration, if your System properties are defined as
-====
- -Dswarm.datasources.data-sources.devwf.connection-url=jdbc:postgresql://localhost:5432/sampledb
- -Dswarm.datasources.data-sources.devwf.driver-name=postgresql
- -Dswarm.datasources.data-sources.devwf.jndiname=java:/jboss/datasources/devwf
- -Dswarm.datasources.data-sources.devwf.user-name=postgres
- -Dswarm.datasources.data-sources.devwf.password=admin
+.An example data source configuration
```
Comment (tradej) [inline docs/reference/configuration.adoc:196]: I took the liberty to reformat the extensive configuration into tables that show the respective system property and environment variable side by side for clarity.

**Comment by tradej** [comment]
Re changes: Don't be too harsh on yourself, it's my job to poke holes in the documentation. :)

Re data source example: I believe both locations for the example are useful because here, it's an example of the actual values used in configs, and in the Data sources chapter, it's an example of a real data source config. Duplication of this magnitude is not a worry.

**Comment by tradej** [comment]
@rareddy, can you please accept this PR so that wildfly-swarm#860 can proceed?

**Comment by rareddy** [comment]
sorry, @tradej did not realize that was against my repo.

**Pull request merged**